src/common/transformations/src/transformations/common_optimizations/convert_nms_gather_path_to_unsigned.cpp

```diff
@@ -60,14 +62,62 @@ class PropagateNMSPath : public pass::MatcherPass {
+                [](std::shared_ptr<ov::op::v8::If> if_node, std::shared_ptr<ov::Model> body, int body_index) {
+                    const auto& params = body->get_parameters();
+                    for (auto input_desc : if_node->get_input_descriptions(body_index)) {
+                        auto param = params[input_desc->m_body_parameter_index];
+                        auto input_node = if_node->input(input_desc->m_input_index).get_source_output().get_node();
+
+                        if (ov::has_nms_selected_indices(input_node)) {
+                            ov::set_nms_selected_indices(param.get());
+                        }
+                    }
+                };
+            auto handle_if_results =
+                [](std::shared_ptr<ov::op::v8::If> if_node, std::shared_ptr<ov::Model> body, int body_index) {
+                    const auto& results = body->get_results();
+                    for (auto output_desc : if_node->get_output_descriptions(body_index)) {
+                        auto result = results[output_desc->m_body_value_index];
+                        const auto& result_inputs = result->input_values();
+                        auto output_node = if_node->output(output_desc->m_output_index).get_node();
+
+                        if (any_of(result_inputs.begin(), result_inputs.end(), [](const Output<Node>& output) {
+                                return ov::has_nms_selected_indices(output.get_node());
+                            })) {
+                            ov::set_nms_selected_indices(output_node);
+                        }
+                    }
+                };
+
             auto node = m.get_match_root();
-            const auto& inputs = node->input_values();
-            if (any_of(inputs.begin(), inputs.end(), [](const Output<Node>& output) {
-                    return ov::has_nms_selected_indices(output.get_node());
-                })) {
-                ov::set_nms_selected_indices(node.get());
+
+            if (ov::is_type<ov::op::v8::If>(node)) {
+                auto if_node = ov::as_type_ptr<ov::op::v8::If>(node);
+
+                auto then_body = if_node->get_then_body();
+                handle_if_params(if_node, then_body, ov::op::v8::If::THEN_BODY_INDEX);
+
+                auto else_body = if_node->get_else_body();
+                handle_if_params(if_node, else_body, ov::op::v8::If::ELSE_BODY_INDEX);
+
+                ov::pass::Manager manager;
+                manager.register_pass<ov::pass::PropagateNMSPath>();
+                manager.run_passes(then_body);
+                manager.run_passes(else_body);
+
+                handle_if_results(if_node, then_body, ov::op::v8::If::THEN_BODY_INDEX);
+                handle_if_results(if_node, else_body, ov::op::v8::If::ELSE_BODY_INDEX);
+            } else {
+                const auto& inputs = node->input_values();
+                if (any_of(inputs.begin(), inputs.end(), [](const Output<Node>& output) {
```

Comment:
it looks like these lines repeat 3 times (in handle_if_params, in handle_if_results, and here) 
can we move it to some helper function?